Modules/System Tests/AL Rest Client/src/ALRestClientTests.Codeunit.al

```diff
@@ -0,0 +1,395 @@
+codeunit 134971 "AL Rest Client Tests"
+{
+    Subtype = Test;
+
+    var
+        Assert: Codeunit "Library Assert";
+
+    [Test]
+    procedure TestGet()
+    var
+        ALRestClient: Codeunit "AL Rest Client";
+        ALHttpResponseMessage: Codeunit "AL Http Response Message";
+        JsonObject: JsonObject;
+    begin
+        // [SCENARIO] Test GET request
+
+        // [GIVEN] An uninitialized AL Rest Client
+        // [WHEN] The Get method is called
+        ALHttpResponseMessage := ALRestClient.Get('https://httpbin.org/get');
```

Comment:
I'm seeing a lot of the hardcoded https://httpbin.org/ endpoint here.
I'm thinking, if httpbin.org is down, or somehow broken or simply blocking high-traffic, the autotests will fail and potentially block the pipeline, is this okay?
Would it make sense to place this baseaddress in a helper-function?
It would make it easier to switch the url, or even have a pragma to switch between a local/remote server.

Reply:
I expect Microsoft to disable this test eventually, also when using a local server. The same is done in the Azure Blob Storage module with tests that use Azurite.

So this is only to be used for manual testing, which is what I do anyway. The httpbin.org service can run as a docker container, but I didn't bother to go that route.

After all, I now have a Client Handler in place, that supports fake responses. However, using it for these tests would mean I'd just test my own fake responses. It is very useful for testing real-life scenarios though.